src/app/core/services/courseService/course.service.ts

```diff
@@ -29,9 +29,21 @@ export class CourseService {
 		return this.http.get(this.courseListUrl)
```

Comment:
I have a suggestion how to improve observable flow here and readability.
1) put filtering logic to separate function `filterOutdatetCourses`
```typescript
private filterOutdatedCourses(courseList: Course[]): Course[]{
		return courseList.filter((course) => {
			const currDate = new Date();
			const courseDate = (new Date(course.startDate));
			if (courseDate < currDate) {
				const timeDiff = Math.abs(currDate.getTime() - courseDate.getTime());
				const diffDays = Math.ceil(timeDiff / (1000 * 3600 * 24));
				console.log(course.title + ' diff days is ' + diffDays);
				if (diffDays > 14 ) { return false; }
			}
			return true;
		})
	}
```
2) use it in `map` method
```typescript
.map((courseList: Course[]) => this.filterOutdatedCourses(courseList))
```
3) And then instead of saving this list and providing to subject, we can switch flow from http response observable to our BehaviorSubject. It is needed because you don't use subscription to http.
```typescript
// add import for operator
import 'rxjs/add/operator/switchMap';
// declare instead of creating BehaviorSubject
public CourseListSource: BehaviorSubject<Course[]>; // = new BehaviorSubject<Course[]>([]);

//add after .map((courseList: Course[]) => this.filterOutdatedCourses(courseList))
.switchMap(filtered => {
    this.CourseListSource = new BehaviorSubject<Course[]>(filtered);
    return this.CourseListSource;
})
```
And we will get in `CourseListSource` our Behavior Subject with filtered data;
4) Then change `CourseList` to `getCourses()` in `coursesList.component.ts
```typescript
this.courseServiceSubscription = this.courseService.getCourses()
						.subscribe((res: Course[]) => {
```